nova_vm/src/ecmascript/builtins/global_object.rs

```diff
@@ -158,6 +159,24 @@ pub fn perform_eval(
         return Ok(x);
     };
 
+    // Nova custom step: If the eval source string is not a heap string,
+    // pad it with whitespace and allocate it on the heap. This makes it safe
+    // (for some definition of "safe") for the potential functions created in
+    // the eval call to keep references to the string buffer.
+    let x = match x {
+        String::String(x) => x,
+        String::SmallString(x) => {
+            // Add 7 whitespace bytes to the end of the eval string. This should
+            // guarantee that the string gets heap-allocated, no question.
+            let data = format!("{}       ", x.as_str());
+            let heap_string_x = String::from_string(agent, data);
+            let String::String(x) = heap_string_x else {
+                unreachable!()
+            };
+            x
+        }
+    };
```

Comment:
**issue:** Off-by-one-error: It's not actually true that adding 7 whitespace bytes will guarantee the string is heap-allocated, you're forgetting the empty string.

**question:** Wouldn't it be better to transmute the string to `&'static str` in this step, and then slice it for small strings so the whitespace isn't parsed?

Reply:
Haha, true :D

Hmm... Probably the extra whitespace shouldn't really matter. But yeah, micro-opt wise could definite slice it.